Vsix/CodeConversion.cs

```diff
@@ -244,6 +259,24 @@ private async Task<ConversionResult> ConvertDocumentUnhandledAsync<TLanguageConv
             return await ProjectConversion.ConvertSingleAsync<TLanguageConversion>(document, new SingleConversionOptions {SelectedTextSpan = selectedTextSpan, AbandonOptionalTasksAfter = await GetAbandonOptionalTasksAfterAsync()}, CreateOutputWindowProgress(), cancellationToken);
         }
 
+        private async IAsyncEnumerable<ConversionResult> ConvertDocumentsUnhandled<TLanguageConversion>(
+            IReadOnlyCollection<string> documentsPath,
+            [EnumeratorCancellation] CancellationToken cancellationToken) where TLanguageConversion : ILanguageConversion, new()
+        {
+            await _outputWindow.WriteToOutputWindowAsync($"Converting {documentsPath.Count} files...", true, true);
+
+            var documentsIds = documentsPath.Select(t => _visualStudioWorkspace.CurrentSolution.GetDocumentIdsWithFilePath(t).FirstOrDefault()).ToList();
+            if (documentsIds.Any(t => t == null)) {
+                await _outputWindow.WriteToOutputWindowAsync("One or more file is not part of a compiling project, using best effort text conversion (less accurate).");
+                var convertedTexts = documentsPath.Select(t => ConvertFileTextAsync<TLanguageConversion>(t, Span.FromBounds(0, 0), cancellationToken));
+                foreach (var convertedText in convertedTexts) yield return await convertedText;
```

Comment:
Ah yeah you needed to return after this loop. Don't worry about it though, I'll fix this now and deduplicate with the function it's copied from if I can.

Reply:
OK fixed. I made some other changes but they ended up not to actually be useful for the scope of this PR, but pushed them anyway, so just ignore those